CHANGES.md

```diff
@@ -46,6 +46,10 @@ OpenSSL 4.0
 
    *Ryan Hooper*
 
+ * The ENGINE feature was removed.
```

Comment:
I think this deserves a bigger entry than just this. We should describe the fact that all `ENGINE_*` APIs have been removed as a result. We should tell users what they should do about it (transition to providers, change their application to not call `ENGINE_*` APIs any more, or possibly use the stubs if necessary).